### PR TITLE
Add sendNotificationMessage hook

### PR DESCRIPTION
**Pull request by dmolineus**
This pull request provides a hook before sending a message.

I'm using the notification center in an application which sends a notification when something happens. This notification creates different messages for different persons.  For example a `productWasOrdered` notifcation would send different messages to the product owner, the user who ordered the product and the system administrator.

The product owner has the option to disable such email notifications in his profile. To respect his user settings the hook is required. 

Alternatively I would have to create different notifications for each receiver and handle the settings when creating the notification. But this would limit the flexibility which the notification center claims to provide.

### CODE REVIEW

**Comment by dmolineus** [comment]
Another advantage is that the tokens could get enriched using the hook. So when creating the notification I only have to pass the `recipientId`. A hook would add the member username, email and so on automatically.

**Comment by Toflar** [comment]
I would add this to 1.3 even though already in RC phase. I don't expect to have 1.4 anytime soon so I think we should accept this. @aschempp, @qzminski: If you agree, I'll merge it.

**Comment by qzminski** [comment]
I'm fine with adding it now.

**Comment by dmolineus** [comment]
Would be great if it will be part of release 1.3.0. Be aware that I've already rebased the branch with the changes in release/1.3.0. as I needed the fixes made there to get it installable. 

If you want to merge this PR onto another branch, feel free to inform me. I can reopen it.

File: README.md [hunk 1]
```diff
@@ -57,4 +57,34 @@ if (null !== $objNotificationCollection) {
         $objNotification->send($arrTokens, $strLanguage); // Language is optional
     }
 }
-```
\ No newline at end of file
+```
+
+## Hooks
+
+If you want to enrich each message being send by some meta data or want to disable some messages being sent, you can
```
Comment (Toflar) [inline README.md]: `each message being sent`

**Comment by Toflar** [comment]
Looks good to me, one little spelling mistake but then I'll merge :)

**Comment by dmolineus** [comment]
Typo got fixed.

**Comment by Toflar** [comment]
Thanks for the hint, unfortunately one does not get any notification when a new commit is pushed to a PR :)

**Comment by dmolineus** [comment]
I know. That's why I also add the comment.

**Pull request merged**